src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/FindCommandParserTest.java

```diff
@@ -1,40 +1,266 @@
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_AMOUNT_RANGE_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_AMOUNT_SEARCH_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_AMOUNT_TO_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_DUPLICATE_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_FROM_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_RANGE_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_SEARCH_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_TO_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_EMPTY_TITLE_KEYPHRASE;
+import static ay2021s1_cs2103_w16_3.finesse.model.category.Category.MESSAGE_EMPTY_CATEGORY;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import java.util.function.Predicate;
 
 import org.junit.jupiter.api.Test;
 
 import ay2021s1_cs2103_w16_3.finesse.logic.commands.FindCommand;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Date;
 import ay2021s1_cs2103_w16_3.finesse.model.transaction.Transaction;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.HasCategoriesPredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.HasExactAmountPredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.InAmountRangePredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.InDateRangePredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.OnExactDatePredicate;
 import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.TitleContainsKeyphrasesPredicate;
 
 public class FindCommandParserTest {
 
+    public static final String AMOUNT_FROM_DESC_BUBBLE_TEA = " " + PREFIX_AMOUNT_FROM + VALID_AMOUNT_BUBBLE_TEA;
+    public static final String AMOUNT_TO_DESC_BUBBLE_TEA = " " + PREFIX_AMOUNT_TO + VALID_AMOUNT_BUBBLE_TEA;
+    public static final String AMOUNT_FROM_DESC_INTERNSHIP = " " + PREFIX_AMOUNT_FROM + VALID_AMOUNT_INTERNSHIP;
+    public static final String AMOUNT_TO_DESC_INTERNSHIP = " " + PREFIX_AMOUNT_TO + VALID_AMOUNT_INTERNSHIP;
+    public static final String DATE_FROM_DESC_BUBBLE_TEA = " " + PREFIX_DATE_FROM + VALID_DATE_BUBBLE_TEA;
+    public static final String DATE_TO_DESC_BUBBLE_TEA = " " + PREFIX_DATE_TO + VALID_DATE_BUBBLE_TEA;
+    public static final String DATE_FROM_DESC_INTERNSHIP = " " + PREFIX_DATE_FROM + VALID_DATE_INTERNSHIP;
+    public static final String DATE_TO_DESC_INTERNSHIP = " " + PREFIX_DATE_TO + VALID_DATE_INTERNSHIP;
```

Comment:
Just to check, is the reason why a space is prepended so that you can chain these strings together? If so, perhaps it would be good to add a comment stating this.

Reply:
Will add a comment highlighting that these strings are similar to those defined in `CommandTestUtil`, but as they are only used in `FindCommandParserTest` they are defined here.